### PR TITLE
stog: init at 0.17.0

### PR DESCRIPTION
**Pull request by thufschmitt**
(and adds other needed packages)

Tested on NixOS (and with `nox-review`).

### CODE REVIEW

File: pkgs/development/ocaml-modules/xtmpl/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchFromGitHub, ocaml, findlib, uutf, sedlex, ppx_tools, js_of_ocaml
+, re }:
+
+if stdenv.lib.versionOlder ocaml.version "4.03"
+then throw "xtmpl not supported for ocaml ${ocaml.version}"
+else
+stdenv.mkDerivation rec {
+  name = "xtmpl${version}";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/xtmpl/default.nix]: Please add a dash here too.

**Review by vbgl — CHANGES_REQUESTED** [review]

**Comment by regnat** [comment]
@vbgl: thanks for the review, I updated tho PR accordingly

File: pkgs/applications/misc/stog/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  version = "0.17.0";
+  src = fetchFromGitHub {
+    owner = "zoggy";
+    repo = "stog";
+    rev = "release-${version}";
+    sha256 = "06fnl3im0rycn05w39adfmm7w4s8l3jrj43h8f8h3b56grh21x0d";
+  };
+
+  buildInputs = [ ocaml ocf ptime uutf uri ppx_blob xtmpl ocaml_lwt higlo camlp4 ];
+  propagatedBuildInputs = [ findlib omd ];
```
Comment (vbgl) [inline pkgs/applications/misc/stog/default.nix]: How did you chose this list of propagated inputs? The `stog` library needs some more.

**Comment by regnat** [comment]
Indeed, I just picked the one needed for the binary. I'll propagate
the ocaml modules needed by the library as well.

**Comment by vbgl** [comment]
Something went wrong when trying to use github’s merge facility. So I cherry-picked individual commits as 2f020cf7c1aba4e2b02f55395f069e8072d75b68, 40158137fd427a8d76eb157962e7dea152605ee7, 89e5963f2ca90ebfe66609a0ccb1a5c0ea6643d1 and 4e9e1bc7ba2a950542b3b330fd14bb4ce6dceb8f.

**Comment by regnat** [comment]
Thanks

**Pull request closed without merging**